code/mob/living/carbon/human/slasher.dm

```diff
@@ -0,0 +1,562 @@
+			if(isnull(src.last_btype) || isnull(src.last_bdna))
+				src.last_btype = src.blood_type
+				src.last_bdna = src.blood_DNA
+			if(!src.trailing_blood)
+				src.tracked_blood = list("bDNA" = src.last_bdna, "btype" = src.last_btype, "count" = INFINITY)
+				src.track_blood()
+				trailing_blood = TRUE
+			else
+				src.tracked_blood = list("bDNA" = src.last_bdna, "btype" = src.last_btype, "count" = 0)
+				trailing_blood = FALSE
+
+		///Handles creating a machete/the circumstances where we DON'T summon it
+		summon_machete()
+			var/list/machetes = list()
+			var/we_hold_it = 0
+			var/mob/living/M = src
+
+			if (M.hasStatus("stunned") || M.hasStatus("weakened") || M.hasStatus("paralysis") || !isalive(M) || M.restrained())
+				boutput(M, __red("Not when you're incapacitated, restrained, or incorporeal."))
+				return 1
+
+			for_by_tcl(K, /obj/item/slasher_machete)
+				if (M.mind && M.mind.key == K.slasher_key)
+					if (K == M.find_in_hand(K))
+						we_hold_it = 1
+						continue
+					if (!(K in machetes))
+						machetes["[K.name] #[length(machetes) + 1] [ismob(K.loc) ? "carried by [K.loc.name]" : "at [get_area(K)]"]"] += K
+
+			switch (length(machetes))
+				if (-INFINITY to 0)
+					if (we_hold_it != 0)
+						boutput(M, __red("You're already holding your machete."))
+						return 1
+					else
+						boutput(M, __red("You summon a new machete to your hands."))
+						var/obj/item/slasher_machete/N = new /obj/item/slasher_machete(get_turf(M))
+						N.slasher_key = M.mind?.key
+						M.put_in_hand_or_drop(N)
+						return 0
+
+				if (1)
+					var/obj/item/slasher_machete/W = machetes[machetes[1]]
+
+					if (!istype(W))
+						boutput(M, __red("You are unable to summon your machete."))
+						return 1
+
+					src.send_machete_to_target(W)
+
+				// There could be multiple, i guess
+				if (2 to INFINITY)
+					var/t1 = input("Please select a machete to summon", "Target Selection", null, null) as null|anything in machetes
+					if (!t1)
+						return 1
+
+					var/obj/item/slasher_machete/K2 = machetes[t1]
+
+					if (!M || !ismob(M) || !isliving(M) || !M.mind)
+						return 1
+					if (!istype(K2))
+						boutput(M, __red("You are unable to summon your machete."))
+						return 1
+					if (M.hasStatus("stunned") || M.hasStatus("weakened") || M.hasStatus("paralysis") || !isalive(M) || M.restrained())
+						boutput(M, __red("Not when you're incapacitated, restrained, or incorporeal."))
+						return 1
+					if (M.mind.key != K2.slasher_key)
+						boutput(M, __red("You are unable to summon your machete."))
+						return 1
+
+					src.send_machete_to_target(K2)
+
+			return 0
+
+		///Actually sending the machete to the Slasher if one exists already
+		send_machete_to_target(var/obj/item/I)
+			if(!istype(I))
+				return
+
+			I.visible_message("<span class='alert'><b>The [I.name] is suddenly warped away!</b></span>")
+			elecflash(I)
+
+			if(ismob(src.loc))
+				src.u_equip(I)
+			if(istype(locate(I), /obj/item/storage))
```

Comment:
Haven't we talked about how `locate(object)` always returns null already?

Reply:
Bit of context:
this is older code from when i got using length() over .len confused with using locate() over .loc (which are two different things completely, as you know)